### ISSUE TITLE
Vehicle/buildings table/database change needed

### ISSUE THREAD

**Issue by fazmond** [DayZMod/DayZ#290]
ok so with the addition of building parts I think vehicles should have their own section in the database ie vehicle_data away from all the other stuff that are currently located in the object_data part of the data base.

the main reason is managing the data base really its very messy having all that stuff in one table and also would make the counting of cars on the server easier for the spawn/clearup scripts.

im currently having to manually add things not to count into the iNumVehExisting . (which was stopping vehicles to spawn as it were also counting the things below) 
for example :=

SELECT COUNT(*)                 #retrieve the amount of already spawned vehicles...
            INTO iNumVehExisting
            FROM Object_DATA 
            WHERE Instance = sInstance
                    AND Classname != '-'
                    AND Classname != 'Hedgehog_DZ'
                    AND Classname != 'Wire_cat1'
                    AND Classname != 'Sandbag1_DZ'
                    AND Classname != 'TrapBear'
                    AND Classname != 'Trap_Cans'
                    AND Classname != 'CamoNet_DZ'  
                                        AND Classname != 'TentStorage'
                    AND Classname != 'TentStorage0'
                    AND Classname != 'TentStorage1'
                    AND Classname != 'TentStorage2'
                    AND Classname != 'TentStorage3'
                    AND Classname != 'TentStorage4'
                    AND Classname != 'DomeTentStorage'
                    AND Classname != 'DomeTentStorage1'
                    AND Classname != 'DomeTentStorage2'
                    AND Classname != 'DomeTentStorage3'
                    AND Classname != 'DomeTentStorage4'
                    AND Classname != 'Sandbag1_DZ'
                    AND Classname != 'StashMedium'  
                    AND Classname != 'StashMedium1'
                    AND Classname != 'StashMedium2'
                    AND Classname != 'StashMedium3'
                    AND Classname != 'StashMedium4'
                    AND Classname != 'StashSmall'
                    AND Classname != 'StashSmall1'
                    AND Classname != 'StashSmall2'
                    AND Classname != 'StashSmall3'
                    AND Classname != 'StashSmall4'
                    AND Classname != 'WoodenFence_base'
                    AND Classname != 'WoodenFence_1_foundation'
                    AND Classname != 'WoodenFence_1_frame'  
                    AND Classname != 'WoodenFence_2'
                    AND Classname != 'WoodenFence_3'
                    AND Classname != 'WoodenFence_4'
                    AND Classname != 'WoodenFence_5'
                    AND Classname != 'WoodenFence_6'
                    AND Classname != 'WoodenFence_7'
                    AND Classname != 'MetalFence_base'
                    AND Classname != 'MetalFence_halfpanel'
                    AND Classname != 'MetalFence_1_foundation'
                    AND Classname != 'MetalFence_2' 
                    AND Classname != 'MetalFence_3' 
                    AND Classname != 'MetalFence_4' 
                    AND Classname != 'MetalFence_5'
                    AND Classname != 'MetalFence_6' 
                    AND Classname != 'MetalFence_7';

And as more building parts are added to the game this table is only going to get more complicated

**Comment by R4Z0R49** [comment]
SELECT COUNT(*) #retrieve the amount of already spawned vehicles...
INTO iNumVehExisting
FROM Object_DATA 
WHERE Instance = sInstance
AND Classname != '-'
AND Classname != 'Hedgehog_DZ'
AND Classname != 'Wire_cat1'
AND Classname != 'Sandbag1_DZ'
AND Classname != 'TrapBear'
AND Classname != 'Trap_Cans'
AND Classname != 'CamoNet_DZ'
AND Classname != 'TentStorage%'
AND Classname != 'DomeTentStorage%'
AND Classname != 'Sandbag1_DZ'
AND Classname != 'StashMedium%'
AND Classname != 'StashSmall%'
AND Classname != 'WoodenFence%'
AND Classname != 'MetalFence%';

Short version. (untested)

However im pretty sure all buildings are tagged with a charID so a simple 

```
    SELECT COUNT(*)     
        INTO iNumVehExisting
        FROM Object_DATA
        WHERE Instance = sInstance
        AND CharacterID = 0;
```

Will work just as well. (this is already in the sql)

**Comment by fazmond** [comment]
yes that works of course I was working with a script someone had taken from a private server, but I still think its still best practise to maybe have them in their own table.

**Comment by R4Z0R49** [comment]
There really no reason to separate them just now if your having trouble trying to view them setup views in mysql for quick access to what you need. Maybe a later time we can move them around as we bring in more stuff.

**Comment by fazmond** [comment]
this is still a good idea and needs a think about again,
